rust/lancedb/src/database/listing/manifest.rs

```diff
@@ -0,0 +1,3991 @@
+    /// Configuration for new tables
+    new_table_config: NewTableConfig,
+    /// Object store wrapper
+    store_wrapper: Option<Arc<dyn WrappingObjectStore>>,
+    /// Read consistency interval for tables
+    read_consistency_interval: Option<Duration>,
+}
+
+impl ManifestListingDatabase {
+    #[allow(clippy::too_many_arguments)]
+    pub async fn new(
+        uri: String,
+        query_string: Option<String>,
+        session: Arc<Session>,
+        manifest_config: ManifestListingDatabaseConfig,
+        object_store: Arc<ObjectStore>,
+        storage_options: HashMap<String, String>,
+        new_table_config: NewTableConfig,
+        store_wrapper: Option<Arc<dyn WrappingObjectStore>>,
+        read_consistency_interval: Option<Duration>,
+    ) -> Self {
+        let manifest_uri =
+            Self::build_object_uri(&uri, MANIFEST_TABLE_NAME, query_string.as_deref()).unwrap();
+        let manifest_dataset = {
+            let read_params = ReadParams {
+                session: Some(session.clone()),
+                ..Default::default()
+            };
+
+            DatasetBuilder::from_uri(&manifest_uri)
+                .with_read_params(read_params)
+                .load()
+                .await
+                .ok()
+                .map(|dataset| {
+                    DatasetConsistencyWrapper::new_latest(
+                        dataset,
+                        manifest_config.read_consistency_interval,
+                    )
+                })
+        };
+
+        Self {
+            uri,
+            query_string,
+            session,
+            manifest_dataset: std::sync::Mutex::new(manifest_dataset),
+            manifest_uri,
+            manifest_config,
+            object_store,
+            storage_options,
+            new_table_config,
+            store_wrapper,
+            read_consistency_interval,
+        }
+    }
+
+    pub fn build_object_id(namespace: &[String], name: &str) -> String {
+        if namespace.is_empty() {
+            name.to_string()
+        } else {
+            let mut id = namespace.join(DELIMITER);
+            id.push_str(DELIMITER);
+            id.push_str(name);
+            id
+        }
+    }
+
+    pub fn parse_object_id(object_id: &str) -> (Vec<String>, String) {
+        let parts: Vec<&str> = object_id.split(DELIMITER).collect();
+        if parts.len() == 1 {
+            (Vec::new(), parts[0].to_string())
+        } else {
+            let namespace = parts[..parts.len() - 1]
+                .iter()
+                .map(|s| s.to_string())
+                .collect();
+            let name = parts[parts.len() - 1].to_string();
+            (namespace, name)
+        }
+    }
+
+    fn build_object_dir_name(
+        &self,
+        name: &str,
+        namespace: &[String],
+        object_type: ObjectType,
+        object_id: &str,
+    ) -> String {
+        if namespace.is_empty()
+            && self.manifest_config.parent_dir_listing_enabled
```

Comment:
I wonder if we want this to be just a configurable option `use_simple_naming_scheme` or something?  Some advanced users like to poke around at object storage themselves and having the folder names map to the table names can make it more obvious what is going on.

Though I suppose the folder name wouldn't track after rename, etc. and there would be duplicates issues with namespaces so maybe not worth the footgun.